frameworklauncher/src/main/java/com/microsoft/frameworklauncher/common/model/ResourceDescriptor.java

```diff
@@ -43,6 +45,10 @@
   @NotNull
   private List<Range> portRanges = new ArrayList<>();
 
+  @Valid
+  @NotNull
+  private Integer portNumber = 0;
```

Comment:
Add validation like:
GpuConsistentValidation

Reply:
gpu count and gpu attribute cannot do consistent validation.  User can set gpu count not set gpu attribute. 

---
In reply to: [167151249](https://github.com/Microsoft/pai/pull/318#discussion_r167151249) [](ancestors = 167151249)